BAU: Use latest verison of common-test-utils lib

adityapahuja opened the pull request:
This change replaces org.apache.commons.lang.reflect.FieldUtils with org.apache.commons.lang3.reflect.FieldUtils because org.apache.commons.lang.reflect.FieldUtils does not exist. It updates AssertJ and Mockito libraries to the latest versions.

Author: @adityapahuja

Wynndow commented:
Do we definitely not have anything that uses saml-lib that runs on java8 anymore?

CharlesIC commented:
The MSA and VSP still have to support Java 8 and I'm pretty sure they both use saml-lib

adityapahuja commented:
Yes, I discussed it with Chris. I will need to make change to common-test-utils to use OpenJDK 8 first before this.

adityapahuja commented:
I have dropped the second commit which removes OpenJDK 8 and updated the gradle file to use latest version of common-test-utils library suitable for OpenJDK 8.
